### PR TITLE
first cut at issue and PR templates for OpenMDAO

### PR DESCRIPTION
**Pull request by robfalck**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+- [ ] Enhancement (POEM: N/A or POEM number at https://github.com/OpenMDAO/POEMs)
+
+### Description
+
+For bugs, explain what is happening and how it differs from the expected behavior.
+For enhancements, a description of the proposed enhancement should be provided by a POEM at https://github.com/OpenMDAO/POEMs
+
+### Example
+
+For bugs, if at all possible, provide a short snipped which reproduces the issue,
```
Comment (naylor-b) [inline .github/ISSUE_TEMPLATE.md]: snipped --> snippet

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+### Summary of Issue
+
+<A brief summary of the issue (used for release notes).>
+
+### Issue Type
+
+- [ ] Bug
+- [ ] Enhancement (POEM: N/A or POEM number at https://github.com/OpenMDAO/POEMs)
```
Comment (swryan) [inline .github/ISSUE_TEMPLATE.md]: Is it intended that the POEM number be in the square brackets or in the parens?

Would one of the following be more clear?  I think I prefer the latter:

```
[ ] Enhancement, POEM #[   ]  (N/A or POEM number at https://github.com/OpenMDAO/POEMs)
```
```
[ ] Enhancement
    POEM #[   ]  (N/A or POEM number at https://github.com/OpenMDAO/POEMs)
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+Summary of PR.
+
+### Related Issues
+
+- Resolves #
+
+### Status
+
+- [ ] Ready for merge
```
Comment (swryan) [inline .github/PULL_REQUEST_TEMPLATE.md]: Seems like there should me more states here... 

What is the status if the the submitter does not consider the PR "Ready for Merge"? 

One or more of these?
```
[ ] Requesting Feedback
[ ] Requesting Review
[ ] Requesting Approval
```

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (robfalck) [reply .github/ISSUE_TEMPLATE.md]: Agreed, I'll make that change.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (robfalck) [reply .github/PULL_REQUEST_TEMPLATE.md]: I would argue that by saying they consider the work complete/ready to merge, the user is basically putting the ball in our court.  All three of those are true if it's considered ready to merge.

Comment (swryan) [reply .github/PULL_REQUEST_TEMPLATE.md]: then why are they submitting at PR at all?

Comment (robfalck) [reply .github/PULL_REQUEST_TEMPLATE.md]: Sometimes you might submit one while working it to utilize travis

**Pull request merged**